mlrun/datastore/targets.py

```diff
@@ -388,6 +458,30 @@ def add_writer_state(
             features=features, timestamp_key=timestamp_key, key_columns=None
         )
 
+        partition_cols = None
+        if self.key_bucketing_number is not None:
+            partition_cols = [
+                "$key"
+                if self.key_bucketing_number == 0
+                else ("$key", self.key_bucketing_number)
+            ]
+        if self.partition_cols:
+            partition_cols = partition_cols or []
+            partition_cols.extend(self.partition_cols)
+        time_partitioning_granularity = self.time_partitioning_granularity
+        if [
```

Comment:
Not sure what you want to do with False & empty string, but using "if not any" here would be more readable (at least to me :))

Reply:
No can do. 0 would be evaluated as false.